istioctl/pkg/waypoint/waypoint.go

```diff
@@ -56,26 +56,22 @@ const (
 )
 
 func Cmd(ctx cli.Context) *cobra.Command {
-	var waypointServiceAccount string
-	makeGatewayName := func(sa string) string {
-		name := sa
-		if name == "" {
-			name = "namespace"
-		}
-		return name
-	}
-	makeGateway := func(forApply bool) *gateway.Gateway {
+	var waypointServiceAccount, waypointName string
+	makeGateway := func(forApply bool) (*gateway.Gateway, error) {
 		ns := ctx.NamespaceOrDefault(ctx.Namespace())
 		if ctx.Namespace() == "" && !forApply {
 			ns = ""
 		}
+		if waypointName == "" {
```

Comment:
I think a default would probably be nice? Especially if we assume a typical user will have one per namespace, having a convention for the name of that could be handy

Reply:
+1 for a default value.

Reply:
rather than throw an error we can just do "no name specified, using `{namespace}-waypoint`" yea

Reply:
Note the class name is appended to the generated things. So if we keep `istio-waypoint` class, `name-waypoint-istio-waypoint` would be the result which is pretty gross.